lib/utils/addEntries.js

```diff
@@ -56,7 +60,17 @@ function addEntries(config, options, server) {
 
     // eslint-disable-next-line no-shadow
     [].concat(config).forEach((config) => {
-      config.entry = prependEntry(config.entry || './src');
+      const webTarget =
+        config.target === 'web' ||
+        config.target === 'webworker' ||
+        config.target == null;
```

Comment:
I am afraid what some developers can have very difficult setup (example testing hmr using jsdom and using webpack, it is very weird, but can be in theory). Maybe we can implement option to allow developers filter configurations (by default modify configuration only for `web`/`webworker` and null)

Reply:
Addressed in latest commit